syntax = "proto3";

option go_package=".;proto";

import "google/protobuf/empty.proto";

service Order {
    rpc CreateOrder(OrderRequest)returns(OrderDetailResponse);
    rpc QueryOrderList(OrderRequest)returns(OrderListResponse);
    rpc QueryOrderDetail(OrderRequest)returns(OrderDetailResponse);
    rpc DelOrder(OrderRequest)returns(google.protobuf.Empty);
    rpc UpdateOrderStatus(OrderStatusRequest)returns(google.protobuf.Empty);
}

message OrderRequest {
    int32 id = 1;
    int32 user_id = 2;
    string name = 3;
    string mobile = 4;
    string address = 5;
    string post = 6;
}

message CreateOrderResponse {
    int32 id = 1;
    string order_sn = 2;
}

message OrderFilterRequest {
    int32 id = 1;
    int32 page = 2;
    int32 limit = 3;
}

message OrderGoodsDetailResponse {
    sint32 order_id = 1;
    sint32 goods_id = 2;
    string goods_name = 3;
    string goods_image = 4;
    float market_price = 5;
    float shop_price = 6;
    int32 nums = 7;
}
message OrderDetailResponse {
    int32 id = 1;
    string order_sn = 2;
    int32 user_id = 3;
    float coupon = 4;
    int32 delivery = 5;
    string signer_name = 6;
    string signer_mobile = 7;
    string signer_address = 8;
    float amount = 9;
    float pay_amount = 10;
    string status = 11;
    string pay_time = 12;
    string post = 13;
    repeated OrderGoodsDetailResponse goods = 14;
}

message OrderListResponse {
    int32 total = 1;
    repeated OrderDetailResponse data = 2;
}

message OrderStatusRequest {
    string order_sn = 1;
    string status = 2;
}
